packages/scandipwa/src/component/MyAccountAddressForm/MyAccountAddressForm.component.js

```diff
@@ -116,15 +116,26 @@ export class MyAccountAddressForm extends FieldForm {
 
     onZipcodeChange = (e) => {
         const { value } = e.currentTarget;
-        const { countryId } = this.state;
+        const { countryId, availableRegions } = this.state;
+
+        getCityAndRegionFromZipcode(countryId, value).then(
+            ([city, regionCode]) => {
+                if (city) {
+                    this.setState({
+                        city
+                    });
+                }
 
-        const city = getCityFromZipcode(countryId, value);
+                if (availableRegions.length > 0 && regionCode) {
+                    const { id: regionId } = availableRegions
+                        .find((r) => r.code.toUpperCase() === regionCode.toUpperCase());
 
-        if (city) {
-            this.setState({
-                city
-            });
-        }
+                    if (regionId) {
+                        this.setState({ regionId });
+                    }
+                }
+            }
+        );
```

Comment:
Prefer using async functions instead of .then chaining.
Also, where promise chaining is not omittable, remember to put namespaces! Is ESLint disabled on your machine?

Reply:
@yeegor 
I used `then` as it is widely used in the project, more widely than async-await. Will try to replace `then`, whenever possible in the future. And I made replacements in code of this PR.

As for the namespaces, I didn't know there was ESLint rule for this in the project until now. And it is not surprising, as the rule is OFF.

https://github.com/scandipwa/scandipwa/blob/master/packages/scandipwa/package.json#L142

Should I do something about it? It is really risky to forget to add a namespace. But if I change `off` to `on`, I get 800+ errors, almost of which say  `Namespace for this node is not valid! Consider changing it to ...`.

Reply:
Enabling this rule is a completely separate task then. For now, it's ok.